akd_core/src/proto/mod.rs

```diff
@@ -169,7 +171,7 @@ impl From<&crate::LayerProof> for specs::types::LayerProof {
         Self {
             label: MessageField::some((&input.label).into()),
             siblings: input.siblings.iter().map(|s| s.into()).collect::<Vec<_>>(),
-            direction: input.direction.map(|i| i as u32),
+            direction: Some(input.direction as u32),
```

Comment:
Should the .proto instead use a u8 here as well, instead of an Option<u32>?

Reply:
Protobuf doesn't support u8 unfortunately :(
https://developers.google.com/protocol-buffers/docs/proto#scalar

Reply:
Maybe we should just make the Direction a native `u32`, instead of `u8`, then? Just to avoid any confusion or potential bugs when dealing with encodings/decodings.